test/functional/connection.test.js

```diff
@@ -70,6 +71,21 @@ describe('Connection - functional', function () {
     }
   });
 
+  it('should only pass one argument (topology and not error) for topology "open" events', function (done) {
+    const configuration = this.configuration;
+    const client = configuration.newClient({ w: 1 }, { maxPoolSize: 1 });
+
+    client.on('topologyOpening', () => {
+      client.topology.on('open', (...args) => {
+        expect(args.length).to.equal(1);
+        expect(args[0]).to.be.instanceOf(Topology);
+        client.close(done);
```

Comment:
as written, this won't execute if the assertions above fail; for that reason, we want to write our clean up steps in an `afterEach` block; for this test, given that the other tests in this file suffer from a similar flaw, I recommend declaring a `let client;` at the top of the describe block, and then assigning to that variable in L76; in this line, you can just call `done()`, and then you need to add an `afterEach(done => {...})` block where you do `if (client) { client.close(done) }`